DR_0011_React-for-paginated-views.md

```diff
@@ -0,0 +1,194 @@
+
+#### Alternative JavaScript frameworks
+
+React is a state-of-the-art JavaScript framework that is used widely. There are various options (Angular, Ember, Vue), which all do not provide any significant advantages over React. React, however, only provides the rendering layer for web applications, so its footprint is limited. It further allows us to build client-side components, which we can reuse later, should we decide to move full single-page app. 
+
+### Architecture description
+
+#### API
+
+The API does not require many changes. 
+
+We already support pagination and we will be able to request specific pages at varying length using `limit` and `offset` query parameters. 
+
+We also support basic search and sorting sorting, which is sufficient to implement client-side searchable and sortable lists. 
+
+Aside from that, we might have to add smaller changes here and there to fully support the requirements for each of the paginated views.
+
+#### Client-side extensions
+
+We will implement a set of React components that will allow us to compose paginated tables that mirror the functionality currently achieved using DataTables.
+
+The React component `PaginatedTable` will be use across all paginated views and will provide the core functionality including search, pagination navigation, setting the page size and column sorting. `PaginatedTable` will be the root component and will host the store and all event handlers. Functionality listed above will be implemented as separate React components. 
+
+##### Configuring `PaginatedTable`
+
+`PaginatedTable` is configurable. Each paginated view has a different API source (where we load the data from), number of columns and template for how each row in the table should be rendered. 
+
+The following properties must be set when initialising `PaginatedTable`:
+
+| Property         | Type              | Description   | 
+| ---------------- | ----------------- | ------------- | 
+| `url`            | `String`          | The API URL where data is loaded from. | 
+| `columns `       | `Array`           | A list of column definitions. The number of elements defines the number of columns in the table. Each element must define the following keys: <br>`label` (the label used in the table head for the column)<br>`field` (the model field, used to apply sorting)<br>`sortable` (identifies if the column is sortable) keys. | 
```

Comment:
This does not include the column width property, which I believe to be custom per table. Not all columns have the same width. I believe that this will need to be configurable.